### PR TITLE
Add benchmarks endpoints to Orion Web API

### PR DESCRIPTION
**Pull request by notoraptor**
# Description

Hi @bouthilx ! I don't know where's the best place to report this work, so I made a PR here.

This PR adds new Orion Web API entries to get benchmarks.

The PR is based on your own branch ( https://github.com/bouthilx/orion/tree/feature/benchmark_webapi ) rebased with `develop` branch and extended with few commits to fix issues.

# Changes

- Add new entries to Orion Web API:
  - `/benchmarks`: get available benchmarks
  - `/benchmarks/:name`: get given benchmark
- Replace parameter `task_num` with `repetitions`

# Checklist

## Tests
- [ ] I added corresponding tests for bug fixes and new features. If possible, the tests fail without the changes
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [ ] I have updated the relevant documentation related to my changes

## Quality
- [ ] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [ ] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the style guidelines (`$ tox -e lint`)

### CODE REVIEW

**Comment by notoraptor** [comment]
Hi @bouthilx ! I added a supplementary about tests: https://github.com/notoraptor/orion/commit/cd0617cf3d92d5aec33e19f1e90a2a3005c46733

There was a TODO about adding `test bad task, bad assessment and bad algorithm (no algorithm)`. But there was already a test for bad task, a test for bad assessment and a test for bad algorithm (see tests just after the TODO). So, I just added a test for `bad algorithm - no algorithm`, and I removed the TODO comments.

There is one remaining TODO from your code here, but I don't know if it must be resolved in this PR: https://github.com/notoraptor/orion/blob/feature/benchmark_webapi_rebased/src/orion/benchmark/__init__.py#L105

File: src/orion/serving/benchmarks_resource.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            assessment=assessment,
+            task=task,
+            algorithms=algorithms,
+        )
+
+        response = build_benchmark_response(benchmark, assessment, task, algorithms)
+        resp.body = json.dumps(response)
+
+
+def _find_latest_versions(experiments):
```
Comment (bouthilx) [inline src/orion/serving/benchmarks_resource.py:53]: Looks like the functions below where copy pasted from experiments_resource but are not used. Sorry, that was me. Could you please remove them? :grimacing:

Comment (notoraptor) [reply src/orion/serving/benchmarks_resource.py]: Done !

Comment (bouthilx) [reply src/orion/serving/benchmarks_resource.py]: Thanks! I think the other functions are not used as well.

**Comment by notoraptor** [comment]
TODO: Check coverage

**Review by bouthilx — APPROVED** [review]
Coverage it good, thanks a lot!!

**Pull request merged**